mediation_service/mediation/prepare_ssp_response.py

```diff
@@ -154,7 +168,49 @@ def __transform_allergy_local_references(ssp_response: dict):
                 ] = patient_list.get(patient_id[1], patient_ref)
 
 
-def __filter_non_allergy_intolerance(ssp_response: dict):
+def _extract_resolved_allergies(ssp_response: dict):
+    """
+    Select resolved allergy intolerance resources from Ended allergies list
+    Return them as a list of Bundle Entry objects
+    """
+    query = parse("`this`.entry[*].resource.resourceType")
+    matches = query.find(ssp_response)
+
+    resolved_allergy_intolerance_entries = []
+
+    for match in matches:
+        if match.value == "List":
+            index = match.full_path.left.left.right.index
+            list_resource = ssp_response["entry"][index]["resource"]
+
+            # search the resource for the snomed code
+            code_query = parse("`this`.code.coding[*].code")
+            code_matches = code_query.find(list_resource)
+
+            for code_match in code_matches:
+                if code_match.value == ENDED_ALLERGIES_LIST_CODE:
+                    contained = list_resource.get("contained")
```

Comment:
Nice use of `.get()` for null-safe dictionary access, are there cases when the resource won't have a `contained` field?